### PR TITLE
Don't access clipboard passively on iOS

### PR DESCRIPTION
**Pull request by justinmc**
## Description

iOS 14 started showing a notification whenever an app accesses the clipboard.  Flutter does this in order to decide whether or not it can show the Paste button in the text selection menu (see https://github.com/flutter/flutter/pull/54902).  As a stopgap solution, this PR removes this functionality on iOS and always shows the Paste button in order to avoid the notification.

Screen Shot 2020-06-25 at 2 54 16 PM

The long-term solution involves using a [new API](https://developer.apple.com/documentation/uikit/uipasteboard/1829416-hasstrings?language=objc) provided by Apple.  We can use this to detect a pasteable string in the clipboard without reading the text itself.  However, the notification will still show when a user presses the paste button.  Progress on this long term solution will continue to be tracked in the issue https://github.com/flutter/flutter/issues/60145.

## Related Issues

https://github.com/flutter/flutter/issues/60145

## Tests

I tested both cupertino and material on iOS to verify that the Paste button is shown despite there being nothing on the clipboard.

## Breaking Change

None.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -1522,6 +1522,18 @@ class ClipboardStatusNotifier extends ValueNotifier<ClipboardStatus> with Widget
...
   /// Check the [Clipboard] and update [value] if needed.
   void update() {
+    // iOS 14 added a notification that appears when an app accesses the
+    // clipboard. To avoid the notification, don't access the clipboard on iOS,
+    // and instead always shown the paste button, even when the clipboard is
+    // empty.
+    // TODO(justinmc): Use the new iOS 14 clipboard API method hasStrings that
+    // won't trigger the notification.
+    // https://github.com/flutter/flutter/issues/60145
+    if (defaultTargetPlatform == TargetPlatform.iOS) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/text_selection.dart:1532]: Use a switch when checking the defaultTargetPlatform.

File: packages/flutter/test/material/text_selection_test.dart [hunk 2]
```diff
@@ -636,5 +636,57 @@ void main() {
     expect(find.text('Cut'), findsOneWidget);
     expect(find.text('Paste'), findsOneWidget);
     expect(find.text('Select all'), findsOneWidget);
-  }, skip: isBrowser);
+  }, skip: isBrowser, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android }));
+
+    // TODO(justinmc): https://github.com/flutter/flutter/issues/60145
```
Comment (goderbauer) [inline packages/flutter/test/material/text_selection_test.dart:641]: nit: indentation

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Pull request merged**